tests/unit/storage_v3_indices.cpp

```diff
@@ -860,9 +873,117 @@ TEST_F(IndexTest, LabelPropertyIndexCreateWithExistingPrimaryKey) {
+
+  // Create index on secondary label
+  EXPECT_TRUE(storage.CreateIndex(label1));
   EXPECT_EQ(storage.ListAllIndices().label.size(), 1);
   EXPECT_EQ(storage.ListAllIndices().label_property.size(), 1);
 }
 
+TEST_F(IndexTest, LabelIndexCreateVertexAndValidate) {
+  {
+    auto acc = storage.Access();
+    EXPECT_EQ(storage.ListAllIndices().label.size(), 0);
+    EXPECT_EQ(storage.ListAllIndices().label_property.size(), 0);
+  }
+  {
+    auto acc = storage.Access();
+
+    // Create vertices with CreateVertexAndValidate
+    for (int i = 0; i < 5; ++i) {
+      auto vertex =
+          acc.CreateVertexAndValidate(primary_label, {label1}, {{primary_property, PropertyValue(primary_key_id++)}});
+      ASSERT_TRUE(vertex.HasValue());
+    }
+    ASSERT_NO_ERROR(acc.Commit());
+  }
+  {
+    EXPECT_TRUE(storage.CreateIndex(label1));
+    {
+      auto acc = storage.Access();
+      EXPECT_THAT(GetPrimaryKeyIds(acc.Vertices(label1, View::OLD), View::OLD), UnorderedElementsAre(0, 1, 2, 3, 4));
+    }
+  }
+  {
+    EXPECT_TRUE(storage.DropIndex(label1));
+    {
+      auto acc = storage.Access();
+      EXPECT_FALSE(acc.LabelIndexExists(label1));
+    }
+    EXPECT_EQ(storage.ListAllIndices().label.size(), 0);
+  }
+  {
+    auto acc = storage.Access();
+    EXPECT_TRUE(storage.CreateIndex(label1));
+    EXPECT_THAT(GetPrimaryKeyIds(acc.Vertices(label1, View::OLD), View::OLD), UnorderedElementsAre(0, 1, 2, 3, 4));
+
+    for (int i = 0; i < 5; ++i) {
+      auto vertex =
+          acc.CreateVertexAndValidate(primary_label, {label1}, {{primary_property, PropertyValue(primary_key_id++)}});
+      ASSERT_TRUE(vertex.HasValue());
+    }
+
+    EXPECT_THAT(GetPrimaryKeyIds(acc.Vertices(label1, View::NEW), View::NEW),
+                UnorderedElementsAre(0, 1, 2, 3, 4, 5, 6, 7, 8, 9));
+    EXPECT_THAT(GetPrimaryKeyIds(acc.Vertices(label1, View::OLD), View::OLD), UnorderedElementsAre(0, 1, 2, 3, 4));
+  }
+}
+
+TEST_F(IndexTest, LabelPropertyIndexCreateVertexAndValidate) {
+  {
+    auto acc = storage.Access();
+    EXPECT_EQ(storage.ListAllIndices().label.size(), 0);
+    EXPECT_EQ(storage.ListAllIndices().label_property.size(), 0);
+  }
+  {
+    auto acc = storage.Access();
+
+    // Create vertices with CreateVertexAndValidate
+    for (int i = 0; i < 5; ++i) {
+      auto vertex = acc.CreateVertexAndValidate(
+          primary_label, {label1},
+          {{primary_property, PropertyValue(primary_key_id++)}, {prop_id, PropertyValue(vertex_id++)}});
+      ASSERT_TRUE(vertex.HasValue());
+    }
+    ASSERT_NO_ERROR(acc.Commit());
+  }
+  {
+    EXPECT_TRUE(storage.CreateIndex(label1, prop_id));
+    {
+      auto acc = storage.Access();
+      EXPECT_THAT(GetPrimaryKeyIds(acc.Vertices(label1, prop_id, View::OLD), View::OLD),
+                  UnorderedElementsAre(0, 1, 2, 3, 4));
+    }
+  }
+  {
+    EXPECT_TRUE(storage.DropIndex(label1, prop_id));
+    {
+      auto acc = storage.Access();
+      EXPECT_FALSE(acc.LabelIndexExists(label1));
```

Comment:
Shouldn't be this `LabelPropertyIndexExists`?
```suggestion
      EXPECT_FALSE(acc.LabelPropertyIndexExists(label1, prop_id));
```